### ISSUE TITLE
f-string-syntax too aggressive

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#5391]
### Bug description

when the code whose value is to be incorporated into a string contains strings, it does not seem possible to use f-strings:

```z.py
"f-string-syntax too aggressive"
print("test: %s" % (",".join(map(str,range(3)))))
```

### Command used

```shell
pylint z.py
```

### Pylint output

```shell
************* Module z
z.py:2: [C0209(consider-using-f-string), ] Formatting a regular string which could be a f-string
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.5
Python 3.9.5 (default, May 11 2021, 08:20:37) 
[GCC 10.3.0]
```

### OS / Environment

```
$ cat /etc/lsb-release
DISTRIB_ID=Pop
DISTRIB_RELEASE=21.04
DISTRIB_CODENAME=hirsute
DISTRIB_DESCRIPTION="Pop!_OS 21.04"
```

**Comment by Pierre-Sassoulas** [comment]
In this case it's possible to do ``print(f"test: { (','.join(map(str,range(3)))}")``, but it's possible to create an example where there is mixed ``'`` and ``"`` in the internal string too and where f-string is not an option without refactor or backslashes.

**Comment by sam-s** [comment]
This was a simplistic example.
In fact I have many nested formats.

**Comment by DanielNoord** [comment]
We have a check for `\` in the `consider-f-string` checker. It shouldn't be too difficult to add something that checks for inner strings and prevents the suggestion from firing.

**Comment by sam-s** [comment]
Here is a more "realistic" example:
```python
dd = {"a":{"b":[1,2,3]}}
print("<%s[%s]>" % ("foo", "; ".join(
    f"{cond}/{len(subjects)}/{sum(map(len,subjects.values()))}"
    for cond,subjects in dd.items())))
```
Thus I think this is a *bug*.

**Comment by Pierre-Sassoulas** [comment]
Consider that even with a small necessary refactor that does one more assignment, fstrings are faster or at least comparable in term of performance.

``b.py``
```python
import time

dd = {"a": {"b": [1, 2, 3]}}


def percent_string(dd):
    return "<%s[%s]>" % (
        "foo",
        "; ".join(
            f"{cond}/{len(subjects)}/{sum(map(len, subjects.values()))}"
            for cond, subjects in dd.items()
        ),
    )


loop = 1000000
start = time.time()
for i in range(loop):
    percent_string(dd)
end = time.time()
print(f"percent string: {end - start}")

```

Result:
```
python3 b.py     
percent string: 0.8760828971862793
```

``c.py``
```python
import time

dd = {"a": {"b": [1, 2, 3]}}


def f_string(dd):
    generator = (
        f"{cond}/{len(subjects)}/{sum(map(len, subjects.values()))}"
        for cond, subjects in dd.items()
    )
    return f"<{'foo'}[{'; '.join(generator)}]>"


loop = 1000000
start = time.time()
for i in range(loop):
    f_string(dd)
end = time.time()
print(f"fstring: {end - start}")

```

Result:
```
python3 c.py
fstring: 0.8488037586212158
```

**Comment by sam-s** [comment]
Of course, any code can be refactored to use f-strings.
However, creating an extra intermediate variable is not always a good style because it pollutes the namespace in the function.
This is not a "minor" change suggested by the warning.

**Comment by JulienPalard** [comment]
I first though it was too aggressive, too, in my case:

        return "{} on {}".format(
             Truncator(self.user.username if self.user else "Anonymous user").chars(30),
             self.exercise.title,
         )

Of course it can be refactored using f-strings, but it would be less readable, one way I could do it is:

        return (
            f"{Truncator(self.user.username if self.user else 'Anonymous user').chars(30)}"
            f" on {self.exercise.title}"
        )

but it still hit the 88th column, and I find it less readable. The good way would be:

        username = Truncator(
            self.user.username if self.user else "Anonymous user"
        ).chars(30)
        return f"{username} on {self.exercise.title}"

Which is OK, and is the one I will use, or maybe even using two intermediate names:

        username = self.user.username if self.user else "Anonymous user"
        truncated_username = Truncator(username).chars(30)
        return f"{truncated_username} on {self.exercise.title}"

In the end I no longer think the checker is too aggressive, but maybe the message could be enhanced, in the case the string length of the `format` arguments is big, by hinting to use intermediate variables?

**Comment by Pierre-Sassoulas** [comment]
> the message could be enhanced, in the case the string length of the format arguments is big, by hinting to use intermediate variables?

Sounds great (with "big" defined as 'bigger than pylint's ``max-line-length`` option' ? Or maybe more than 2 lines long ?).

**Comment by JulienPalard** [comment]
> more than 2 lines long

This looks like a good definition of "big" to me, yes.

**Comment by sam-s** [comment]
Creating an intermediate variable is not always a good idea.
It might be acceptable in, say, `__str__`, but *not* in a larger function where they would pollute the namespace.

**Comment by Pierre-Sassoulas** [comment]
I think a reasonable compromise is that the message can be disabled on a case by case basis, or a refactor can be done if someone do not want to pollute the namespace. I added this in the better error message project.
